runtime/oti/j9nonbuilder.h

```diff
@@ -5304,8 +5368,8 @@ typedef struct J9VMThread {
 
 #define J9VMTHREAD_REFERENCE_SIZE(vmThread) (J9VMTHREAD_COMPRESS_OBJECT_REFERENCES(vmThread) ? sizeof(U_32) : sizeof(UDATA))
 #define J9VMTHREAD_OBJECT_HEADER_SIZE(vmThread) (J9VMTHREAD_COMPRESS_OBJECT_REFERENCES(vmThread) ? sizeof(J9ObjectCompressed) : sizeof(J9ObjectFull))
-#define J9VMTHREAD_CONTIGUOUS_HEADER_SIZE(vmThread) (J9VMTHREAD_COMPRESS_OBJECT_REFERENCES(vmThread) ? sizeof(J9IndexableObjectContiguousCompressed) : sizeof(J9IndexableObjectContiguousFull))
-#define J9VMTHREAD_DISCONTIGUOUS_HEADER_SIZE(vmThread) (J9VMTHREAD_COMPRESS_OBJECT_REFERENCES(vmThread) ? sizeof(J9IndexableObjectDiscontiguousCompressed) : sizeof(J9IndexableObjectDiscontiguousFull))
+#define J9VMTHREAD_CONTIGUOUS_HEADER_SIZE(vmThread) (((J9VMThread *)vmThread)->contiguousIndexableHeaderSize)
+#define J9VMTHREAD_DISCONTIGUOUS_HEADER_SIZE(vmThread) (((J9VMThread *)vmThread)->discontiguousIndexableHeaderSize)
```

Comment:
Now that we cache the fields in VMThread, these macros become less useful. I'd remove them (I just like simple C expression with -> operator), like we removed _VM variant (but I'll let @dmitripivkine voice his opinion, too).

If we decide to keep, I think:
- if we keep one set (thread) we should keep the other set (vm), too.
- we could also expand the names to include _INDEXIBLE_ (like we did with field names)?
- I would not use this cast: (J9VMThread *)vmThread.  vmThread better be of that type already - if not I want compiler tell me that, so we make sure the caller does the cast

Reply:
My choice would be to keep macro as a generalized way to get header size. The rational (obvious) behind this no need to patch VM code if it is required to be modified in the future again

Reply:
Ok, sounds good, I have just pushed commits that address this conversation. `vmThread` macros have been left in and `JavaVM` macros have been re-introduced. Renaming to add `_INDEXIBLE_` in the macros is also completed, and `(J9VMThread *)vmThread` casting has been removed in the `vmThread` macro.